org.eclipse.xtext/src/org/eclipse/xtext/validation/ConfigurableIssueCodesProvider.java

```diff
@@ -7,22 +7,53 @@
  *******************************************************************************/
 package org.eclipse.xtext.validation;
 
-import java.util.Collections;
 import java.util.Map;
 
 import org.eclipse.xtext.preferences.PreferenceKey;
+import org.eclipse.xtext.util.IAcceptor;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
 
 /**
  * @author Sven Efftinge - Initial contribution and API
  * @since 2.4
  */
 public class ConfigurableIssueCodesProvider {
+	
+	private Map<String, PreferenceKey> issueCodes;
 
+	public ConfigurableIssueCodesProvider() {
+		final Map<String, PreferenceKey> map = Maps.newLinkedHashMap();
+		initialize(new IAcceptor<PreferenceKey>() {
```

Comment:
I wonder if there is a use-case where we would want to use another acceptor. Maybe a factory method should be present, e.g. the acceptor would be obtained via `createAcceptor(map)`

Reply:
we did not need that in the project where we customized this. we simpy did the copy&paste that is now no longer needed

Reply:
ok